src/middlewared/middlewared/plugins/vm/lifecycle.py

```diff
@@ -142,13 +142,16 @@ async def __event_system_ready(middleware, event_type, args):
     Method called when system is ready, supposed to start VMs
     flagged that way.
     """
+
     async def stop_vm(mw, vm):
         stop_job = await mw.call('vm.stop', vm['id'], {'force_after_timeout': True})
         await stop_job.wait()
         if stop_job.error:
             mw.logger.error(f'Stopping VM {vm["name"]} failed: {stop_job.error}')
 
-    if args['id'] == 'ready':
+    # we ignore the 'ready' event on an HA system since the failover event plugin
+    # is responsible for starting this service
+    if args['id'] == 'ready' and not await middleware.call('failover.licensed'):
```

Comment:
We still need to initialize vms, let's please remove this condition as it's already being done below at L159